service/src/test/java/uk/gov/hmcts/reform/fpl/service/CreateC21OrderServiceTest.java

```diff
@@ -0,0 +1,167 @@
+        CaseData caseData = addC21OrderAndBundleToCaseData(emptyC21OrderBundle, "C21_1.pdf");
+
+        List<Element<C21OrderBundle>> c21OrderBundleWithOrder = createC21OrderService.addToC21OrderBundle(
+            caseData.getTemporaryC21Order(), caseData.getJudgeAndLegalAdvisor(), caseData.getC21OrderBundle());
+        assertThat(c21OrderBundleWithOrder).size().isEqualTo(1);
+
+        C21OrderBundle c21OrderBundle = c21OrderBundleWithOrder.get(0).getValue();
+
+        assertThat(c21OrderBundle.getC21OrderDocument().getFilename()).isEqualTo("C21_1.pdf");
+        assertThat(c21OrderBundle.getOrderTitle()).isEqualTo("Example order title");
+        assertThat(c21OrderBundle.getJudgeTitleAndName()).isEqualTo("His Honour Judge Johnson");
+    }
+
+    @Test
+    void shouldAppendNewC21OrderToExistingC21OrderBundle() {
+        CaseData caseData = addC21OrderAndBundleToCaseData(getExistingC21OrderBundle(), "C21_2.pdf");
+
+        List<Element<C21OrderBundle>> c21OrderBundleWithTwoOrders = createC21OrderService.addToC21OrderBundle(
+            caseData.getTemporaryC21Order(), caseData.getJudgeAndLegalAdvisor(), caseData.getC21OrderBundle());
+        assertThat(c21OrderBundleWithTwoOrders).size().isEqualTo(2);
+
+        C21OrderBundle previousC21 = c21OrderBundleWithTwoOrders.get(0).getValue();
+        C21OrderBundle appendedC21 = c21OrderBundleWithTwoOrders.get(1).getValue();
+
+        assertThat(previousC21.getC21OrderDocument().getFilename()).isEqualTo("C21_1.pdf");
+        assertThat(appendedC21.getC21OrderDocument().getFilename()).isEqualTo("C21_2.pdf");
+    }
+
+    @Test
+    void shouldFormatC21TemplateDataCorrectlyWhenOnlyC21MandatoryDataIsIncluded() {
+        CaseData caseData = CaseData.builder()
+            .caseLocalAuthority("example")
+            .familyManCaseNumber("123")
+            .temporaryC21Order(C21Order.builder()
+                .orderDetails("Example order details")
+                .build())
+            .build();
+
+        Map<String, Object> templateData = createC21OrderService.getC21OrderTemplateData(caseData);
+
+        assertThat(templateData.get("courtName")).isEqualTo("Example Court");
+        assertThat(templateData.get("familyManCaseNumber")).isEqualTo("123");
+        assertThat(templateData.get("orderTitle")).isEqualTo("Order");
+        assertThat(templateData.get("orderDetails")).isEqualTo("Example order details");
+        assertThat(templateData.get("todaysDate")).isEqualTo(dateFormatterService.formatLocalDateToString(TODAYS_DATE,
+            FormatStyle.LONG));
+        assertThat(templateData.get("judgeTitleAndName")).isEqualTo("");
+        assertThat(templateData.get("legalAdvisorName")).isEqualTo("");
+        assertThat(templateData.get("children")).isEqualTo(ImmutableList.of());
+    }
+
+    @Test
+    void shouldFormatC21TemplateDataCorrectlyWhenC21TemplateDataIsPopulated() {
+        CaseData caseData = CaseData.builder()
+            .caseLocalAuthority("example")
```

Comment:
we could extract the building of case data to a private method here, what do you think?